### ISSUE TITLE
[BUG] Resetting winget sources throws error message

### ISSUE THREAD

**Issue by llvs** [marticliment/UniGetUI#2544]
### Please confirm these before moving forward

- [X] I have searched for my issue and have not found a work-in-progress/duplicate/resolved issue.
- [X] I have tested that this issue has not been fixed in the latest [(beta or stable) release](https://github.com/marticliment/WingetUI/releases/).
- [X] I have checked the [FAQ](https://github.com/marticliment/WingetUI#frequently-asked-questions) section for solutions.
- [X] This issue is about a bug (if it is not, please use the [correct template](https://github.com/marticliment/WingetUI/issues/new/choose)).

### UniGetUI Version

3.1.1-beta3

### Windows version, edition, and architecture

Win 10 22H2 19045.4651 x64

### Describe your issue

Upon trying to reset the winget sources, the following error is thrown:
```
Operation failed: Der Bereitstellungsvorgang "Remove" mit Zielvolume "C:" für das Paket "Microsoft.Winget.Source_2024.704.2449.57_neutral__8wekyb3d8bbwe" von " " ist mit dem Fehler 0x80073D19 fehlgeschlagen. Weitere Informationen zum Diagnostizieren von Problemen bei der Bereitstellung von Apps finden Sie unter "http://go.microsoft.com/fwlink/?LinkId=235160".
0x80073d19 : Fehler aufgrund der Abmeldung eines Benutzers
```
My system uses a normal user and an admin user which still creates problems for UniGetUI. While I can use CLI winget and have all my software and updates shown, UniGetUI does not succeed (neither using system WinGet nor packaged WinGet).
I think it started with version 3.0.2 but I am not sure as I regularly try all the latest UniGetUI versions on two machines.

### UniGetUI Log

```text
[26.07.2024 11:37:02]    __  __      _ ______     __  __  ______
                        / / / /___  (_) ____/__  / /_/ / / /  _/
                       / / / / __ \/ / / __/ _ \/ __/ / / // /
                      / /_/ / / / / / /_/ /  __/ /_/ /_/ // /
                      \____/_/ /_/_/\____/\___/\__/\____/___/
                          Welcome to UniGetUI Version 3.1.1-beta3
[26.07.2024 11:37:03]
[26.07.2024 11:37:03] Version Code:  3,1093
[26.07.2024 11:37:03] System.FormatException: The input string '850.' was not in a correct format.
                         at System.Number.ThrowFormatException[TChar](ReadOnlySpan`1 value)
                         at System.Int32.Parse(String s)
                         at UniGetUI.Core.Data.CoreData.GetCodePage() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.Core.Data\CoreData.cs:line 26
[26.07.2024 11:37:03] Encoding Code Page set to 0
[26.07.2024 11:37:03] Using bundled GSudo at C:\Program Files\WingetUI\Assets\Utilities\gsudo.exe
[26.07.2024 11:37:03] Loaded language locale: de
[26.07.2024 11:37:04] Randomly-generated background API auth token for the current session: 5e2bsko9jqm3i23szv8gexfw7d17xai7zhpwiqxx5xr0opx462kcrss4bw9zj0jl
[26.07.2024 11:37:04] Api running on http://localhost:7058
[26.07.2024 11:37:04] Lang files were updated successfully from GitHub
[26.07.2024 11:37:04] UniGetUI is up to date
[26.07.2024 11:37:05] Command pwsh.exe was not found on the system
[26.07.2024 11:37:05] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: PowerShell7
                      █ Enabled: True
                      █ Found: False
                      █ THE MANAGER WAS NOT FOUND. PERHAPS IT IS NOT 
                      █ INSTALLED OR IT HAS BEEN MISCONFIGURED 
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[26.07.2024 11:37:05] Command npm was not found on the system
[26.07.2024 11:37:05] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Npm
                      █ Enabled: False
                      █ THE MANAGER IS DISABLED
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[26.07.2024 11:37:05] Downloaded new icons and screenshots successfully!
[26.07.2024 11:37:05] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: .NET Tool
                      █ Enabled: False
                      █ THE MANAGER IS DISABLED
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[26.07.2024 11:37:05] Error finding sources for manager Pip
[26.07.2024 11:37:05] System.InvalidOperationException: Manager Pip does not support custom sources but yet GetSources method was called.
                          at System.Environment.get_StackTrace()
                         at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.AssertSourceCompatibility(String MethodName) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.PackageManagerClasses\Manager\PackageManager.cs:line 354
                         at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.GetSources() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.PackageManagerClasses\Manager\PackageManager.cs:line 399
                         at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
                         at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.GetSources()
                         at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.InitializeAsync() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.PackageManagerClasses\Manager\PackageManager.cs:line 91
                         at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s)
                         at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
                         at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread)
                         at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext()
                         at System.Threading.Tasks.AwaitTaskContinuation.RunCallback(ContextCallback callback, Object state, Task& currentTask)
                         at System.Threading.Tasks.Task.RunContinuations(Object continuationObject)
                         at System.Threading.Tasks.Task`1.TrySetResult(TResult result)
                         at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetExistingTaskResult(Task`1 task, TResult result)
                         at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetResult(TResult result)
                         at UniGetUI.PackageEngine.Managers.PipManager.Pip.LoadManager() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.Pip\Pip.cs:line 360
                         at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s)
                         at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
                         at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread)
                         at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext()
                         at Microsoft.UI.Dispatching.DispatcherQueueSynchronizationContext.<>c__DisplayClass2_0.<Post>b__0()
                         at ABI.Microsoft.UI.Dispatching.DispatcherQueueHandler.Do_Abi_Invoke(IntPtr thisPtr)
                         at ABI.Microsoft.UI.Xaml.IApplicationStaticsMethods.Start(IObjectReference _obj, ApplicationInitializationCallback callback)
                         at Microsoft.UI.Xaml.Application.Start(ApplicationInitializationCallback callback)
                         at UniGetUI.EntryPoint.AsyncMain(String[] args) in C:\SomePrograms\WingetUI-Store\src\UniGetUI\EntryPoint.cs:line 71
                         at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
                         at UniGetUI.EntryPoint.AsyncMain(String[] args)
                         at UniGetUI.EntryPoint.Main(String[] args) in C:\SomePrograms\WingetUI-Store\src\UniGetUI\EntryPoint.cs:line 32
                         at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.AssertSourceCompatibility(String MethodName) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.PackageManagerClasses\Manager\PackageManager.cs:line 354
                         at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.GetSources() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.PackageManagerClasses\Manager\PackageManager.cs:line 399
[26.07.2024 11:37:05] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Pip
                      █ Enabled: True
                      █ Found: True
                      █ Fancye exe name: pip
                      █ Executable path: C:\Users\[myusername]\AppData\Local\Microsoft\WindowsApps\python.exe
                      █ Call arguments:  -m pip
                      █ Version: 
                      █   
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[26.07.2024 11:37:06] Command scoop was not found on the system
[26.07.2024 11:37:06] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Scoop
                      █ Enabled: False
                      █ THE MANAGER IS DISABLED
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[26.07.2024 11:37:07] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Winget
                      █ Enabled: True
                      █ Found: True
                      █ Fancye exe name: winget.exe
                      █ Executable path: C:\Users\[myusername]\AppData\Local\Microsoft\WindowsApps\winget.exe
                      █ Call arguments: 
                      █ Version: 
                      █   System WinGet CLI Version: v1.8.1911
                      █   Microsoft.WinGet.Client PSModule version: ""
                      █   Using Native WinGet helper (COM Api)
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[26.07.2024 11:37:07] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: Chocolatey
                      █ Enabled: False
                      █ THE MANAGER IS DISABLED
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[26.07.2024 11:37:08] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
                      █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
                      █ Name: PowerShell
                      █ Enabled: True
                      █ Found: True
                      █ Fancye exe name: powershell.exe
                      █ Executable path: C:\Windows\system32\windowspowershell\v1.0\powershell.exe
                      █ Call arguments:  -NoProfile -Command
                      █ Version: 
                      █   Name                           Value                                                                                   
                      █   ----                           -----                                                                                   
                      █   PSVersion                      5.1.19041.4648                                                                          
                      █   PSEdition                      Desktop                                                                                 
                      █   PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                 
                      █   BuildVersion                   10.0.19041.4648                                                                         
                      █   CLRVersion                     4.0.30319.42000                                                                         
                      █   WSManStackVersion              3.0                                                                                     
                      █   PSRemotingProtocolVersion      2.3                                                                                     
                      █   SerializationVersion           1.1.0.1
                      ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
[26.07.2024 11:37:09] Resetting administrator rights cache for process id 16676
[26.07.2024 11:37:09] Microsoft.Powershell.Client detection has been forcefully skipped as the module is not required on the current context
[26.07.2024 11:37:09] Dependency WinGet PowerShell Module for manager Winget is present
[26.07.2024 11:37:09] LoadComponentsAsync finished executing. All managers loaded. Proceeding to interface.
[26.07.2024 11:37:09] Found 0 available updates from Pip
[26.07.2024 11:37:09] Found 0 installed packages from Pip
[26.07.2024 11:37:12] Found 1 installed packages from PowerShell
[26.07.2024 11:37:14] Found 0 available updates from PowerShell
[26.07.2024 11:37:16] Error finding installed packages on manager Winget
[26.07.2024 11:37:16] System.InvalidOperationException: WinGet: Failed to connect to composite catalog.
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.GetLocalWinGetPackages(INativeTaskLogger logger) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 199
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.<>c__DisplayClass5_0.<GetInstalledPackages_UnSafe>b__0() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 164
                         at System.Threading.Tasks.Task`1.InnerInvoke()
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                      --- End of stack trace from previous location ---
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
                      --- End of stack trace from previous location ---
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.GetInstalledPackages_UnSafe(WinGet Manager) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 164
                         at UniGetUI.PackageEngine.Managers.WingetManager.WinGet.<GetInstalledPackages_UnSafe>b__34_0() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGet.cs:line 137
                         at System.Threading.Tasks.Task`1.InnerInvoke()
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                      --- End of stack trace from previous location ---
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
                      --- End of stack trace from previous location ---
                         at UniGetUI.PackageEngine.Managers.WingetManager.WinGet.GetInstalledPackages_UnSafe() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGet.cs:line 137
                         at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.GetInstalledPackages() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.PackageManagerClasses\Manager\PackageManager.cs:line 228
[26.07.2024 11:37:18] Error finding updates on manager Winget
[26.07.2024 11:37:18] System.InvalidOperationException: WinGet: Failed to connect to composite catalog.
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.GetLocalWinGetPackages(INativeTaskLogger logger) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 199
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.<>c__DisplayClass4_0.<GetAvailableUpdates_UnSafe>b__0() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 144
                         at System.Threading.Tasks.Task`1.InnerInvoke()
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                      --- End of stack trace from previous location ---
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
                      --- End of stack trace from previous location ---
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.GetAvailableUpdates_UnSafe(WinGet Manager) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 144
                         at UniGetUI.PackageEngine.Managers.WingetManager.WinGet.<GetAvailableUpdates_UnSafe>b__33_0() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGet.cs:line 132
                         at System.Threading.Tasks.Task`1.InnerInvoke()
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                      --- End of stack trace from previous location ---
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
                      --- End of stack trace from previous location ---
                         at UniGetUI.PackageEngine.Managers.WingetManager.WinGet.GetAvailableUpdates_UnSafe() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGet.cs:line 132
                         at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.GetAvailableUpdates() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.PackageManagerClasses\Manager\PackageManager.cs:line 201
[26.07.2024 11:37:29] Found 0 available updates from Pip
[26.07.2024 11:37:35] Found 0 available updates from PowerShell
[26.07.2024 11:37:35] Error finding updates on manager Winget
[26.07.2024 11:37:35] System.InvalidOperationException: WinGet: Failed to connect to composite catalog.
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.GetLocalWinGetPackages(INativeTaskLogger logger) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 199
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.<>c__DisplayClass4_0.<GetAvailableUpdates_UnSafe>b__0() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 144
                         at System.Threading.Tasks.Task`1.InnerInvoke()
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                      --- End of stack trace from previous location ---
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
                      --- End of stack trace from previous location ---
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.GetAvailableUpdates_UnSafe(WinGet Manager) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 144
                         at UniGetUI.PackageEngine.Managers.WingetManager.WinGet.<GetAvailableUpdates_UnSafe>b__33_0() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGet.cs:line 132
                         at System.Threading.Tasks.Task`1.InnerInvoke()
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                      --- End of stack trace from previous location ---
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
                      --- End of stack trace from previous location ---
                         at UniGetUI.PackageEngine.Managers.WingetManager.WinGet.GetAvailableUpdates_UnSafe() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGet.cs:line 132
                         at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.GetAvailableUpdates() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.PackageManagerClasses\Manager\PackageManager.cs:line 201
[26.07.2024 11:37:42] Found 0 installed packages from Pip
[26.07.2024 11:37:46] Found 1 installed packages from PowerShell
[26.07.2024 11:37:52] Error finding installed packages on manager Winget
[26.07.2024 11:37:52] System.InvalidOperationException: WinGet: Failed to connect to composite catalog.
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.GetLocalWinGetPackages(INativeTaskLogger logger) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 199
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.<>c__DisplayClass5_0.<GetInstalledPackages_UnSafe>b__0() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 164
                         at System.Threading.Tasks.Task`1.InnerInvoke()
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                      --- End of stack trace from previous location ---
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
                      --- End of stack trace from previous location ---
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.GetInstalledPackages_UnSafe(WinGet Manager) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 164
                         at UniGetUI.PackageEngine.Managers.WingetManager.WinGet.<GetInstalledPackages_UnSafe>b__34_0() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGet.cs:line 137
                         at System.Threading.Tasks.Task`1.InnerInvoke()
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                      --- End of stack trace from previous location ---
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
                      --- End of stack trace from previous location ---
                         at UniGetUI.PackageEngine.Managers.WingetManager.WinGet.GetInstalledPackages_UnSafe() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGet.cs:line 137
                         at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.GetInstalledPackages() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.PackageManagerClasses\Manager\PackageManager.cs:line 228
[26.07.2024 11:49:19] Found 0 installed packages from Pip
[26.07.2024 11:49:23] Found 1 installed packages from PowerShell
[26.07.2024 11:49:27] Error finding installed packages on manager Winget
[26.07.2024 11:49:27] System.InvalidOperationException: WinGet: Failed to connect to composite catalog.
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.GetLocalWinGetPackages(INativeTaskLogger logger) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 199
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.<>c__DisplayClass5_0.<GetInstalledPackages_UnSafe>b__0() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 164
                         at System.Threading.Tasks.Task`1.InnerInvoke()
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                      --- End of stack trace from previous location ---
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
                      --- End of stack trace from previous location ---
                         at UniGetUI.PackageEngine.Managers.WingetManager.NativeWinGetHelper.GetInstalledPackages_UnSafe(WinGet Manager) in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGetHelpers\NativeWinGetHelper.cs:line 164
                         at UniGetUI.PackageEngine.Managers.WingetManager.WinGet.<GetInstalledPackages_UnSafe>b__34_0() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGet.cs:line 137
                         at System.Threading.Tasks.Task`1.InnerInvoke()
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                      --- End of stack trace from previous location ---
                         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
                         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
                      --- End of stack trace from previous location ---
                         at UniGetUI.PackageEngine.Managers.WingetManager.WinGet.GetInstalledPackages_UnSafe() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.WinGet\WinGet.cs:line 137
                         at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.GetInstalledPackages() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.PackageManagerClasses\Manager\PackageManager.cs:line 228
[26.07.2024 11:52:27] Found 0 installed packages from Pip
[26.07.2024 11:52:30] Found 1 installed packages from PowerShell
```

### Package Managers Logs

```text
UniGet UI throws an error getting the logs.
```

**Comment by marticliment** [comment]
You are facing an issue with a corrupted WinGet installation.
Please disable WinGet COM API from UniGetUI Settings -> WinGet, and perhaps report this issue to microsoft/winget-cli.

**Comment by ratepenado** [comment]
@llvs I solved a similar issue by deleting the folder WinGet in %LOCALAPPDATA%\Temp and restarting UnigetUI

**Comment by llvs** [comment]
Thank you. My issue seems to be like in the solution described in this 
[winget issue](https://github.com/microsoft/winget-cli/issues/2930#issuecomment-2015746587) (and following post).
Deleting the content of the folder was not enough, the issue is that the folder is an admin rights folder in the user's AppData folder. Another poster in that issue mentions that this has probably happened by usage of gsudo (which is also used by UniGetUI). I also use the non-recommended option of UniGetUI to only ask once for permission because this is the main appeal of using UniGetUI at all: Being able to update everything with only one admin prompt.

**Comment by benjo-90** [comment]
@llvs Today I did a trial run of UniGetUI to see if it could replace my use of winget-cli. I mainly use a non-admin user account, so I have to confirm admin rights via the UAC prompt to update/install some packages using winget-cli. 
UniGetUI was able to update almost all packages (even some that required admin rights), but failed for the VLC player, for example. Here I get a UAC prompt for gsudo and after granting permission, I lose permissions for the WinGet dir in \AppData\Local\Temp with my non-admin account. Afterwards, UniGetUI and winget-cli no longer work with my non-admin account.

UniGetUI and winget-cli worked again after I gave my non-admin account again access to the WinGet folder.
Now, when I try to update VLC in UniGetUI, it again asks for gsudo through a UAC prompt, and when I grant the permission, UniGetUI and winget-cli stop working again, and so on...

I am not sure, but VLC is different from the other packages in a way that if I want to upgrade VLC using winget-cli, it will tell me that I need to uninstall the current version and install the latest version. Maybe this difference is part of the reason for this bug.
